File: src/net/eduard/api/core/BukkitReplacers.java
package net.eduard.api.core;

import org.bukkit.Statistic;
import org.bukkit.entity.Damageable;
import org.bukkit.entity.HumanEntity;
import org.bukkit.entity.Player;

import net.eduard.api.lib.modules.Mine;
import net.eduard.api.lib.modules.Extra;
import net.eduard.api.lib.modules.VaultAPI;

/**
 * Registrando os replacers mais usados na Displayboard
 * 
 * @author Eduard
 *
 */
public class BukkitReplacers {

	public BukkitReplacers() {

		if (Mine.hasPlugin("Vault")) {
			Mine.addReplacer("$player_group", p -> VaultAPI.getPermission().getPrimaryGroup(p));
			Mine.addReplacer("$player_prefix", p -> Mine.toChatMessage(VaultAPI.getChat().getPlayerPrefix(p)));
			Mine.addReplacer("$player_suffix", p -> Mine.toChatMessage(Mine.toChatMessage(VaultAPI.getChat().getPlayerPrefix(p))));
			Mine.addReplacer("$group_prefix", p -> Mine.toChatMessage(
					VaultAPI.getChat().getGroupPrefix("null", VaultAPI.getPermission().getPrimaryGroup(p))));
			Mine.addReplacer("$group_suffix", p -> Mine.toChatMessage(
					VaultAPI.getChat().getGroupSuffix("null", VaultAPI.getPermission().getPrimaryGroup(p))));
			Mine.addReplacer("$player_money", p -> {
				if (VaultAPI.hasVault() && VaultAPI.hasEconomy()) {

					return Extra.MONEY.format(VaultAPI.getEconomy().getBalance(p));

				}
				return "0.00";
			});
			Mine.addReplacer("$player_balance", p -> {
				if (VaultAPI.hasVault() && VaultAPI.hasEconomy()) {

					return Extra.formatMoney(VaultAPI.getEconomy().getBalance(p));

				}
				return "0.00";
			});
		}

		Mine.addReplacer("$players_online", p -> Mine.getPlayers().size());
		Mine.addReplacer("$player_world", p -> p.getWorld().getName());
		Mine.addReplacer("$player_displayname", Player::getDisplayName);
		Mine.addReplacer("$player_name", HumanEntity::getName);
		Mine.addReplacer("$player_health", Damageable::getHealth);
		Mine.addReplacer("$player_maxhealth", Damageable::getMaxHealth);
		Mine.addReplacer("$player_level", Player::getLevel);
		Mine.addReplacer("$player_xp", p -> Extra.MONEY.format(p.getTotalExperience()));
		Mine.addReplacer("$player_kills", p -> p.getStatistic(Statistic.PLAYER_KILLS));
		Mine.addReplacer("$player_deaths", p -> p.getStatistic(Statistic.DEATHS));
		Mine.addReplacer("$player_kdr", p -> {
			int kill = p.getStatistic(Statistic.PLAYER_KILLS);
			int death = p.getStatistic(Statistic.DEATHS);
			if (kill == 0)
				return 0;
			if (death == 0)
				return 0;
			return kill / death;
		});
		Mine.addReplacer("$player_kill/death", p -> {
			int kill = p.getStatistic(Statistic.PLAYER_KILLS);
			int death = p.getStatistic(Statistic.DEATHS);
			if (kill == 0)
				return 0;
			if (death == 0)
				return 0;
			return kill / death;
		});

		Mine.addReplacer("$player_x", p -> p.getLocation().getX());
		Mine.addReplacer("$player_y", p -> p.getLocation().getY());
		Mine.addReplacer("$player_z", p -> p.getLocation().getZ());

	}

	// Strings
	

	
}

File: src/main/java/net/eduard/api/core/BukkitReplacers.kt
package net.eduard.api.core

import net.eduard.api.lib.kotlin.format
import net.eduard.api.lib.modules.Extra
import net.eduard.api.lib.modules.Mine
import net.eduard.api.lib.modules.VaultAPI

import org.bukkit.Statistic

/**
 * Registrando os replacers mais usados na Displayboard
 *
 * @author Eduard
 */
class BukkitReplacers {
    init {
        if (Mine.hasPlugin("Vault")) {
            Mine.addReplacer("player_group") { VaultAPI.getPermission().getPrimaryGroup(it) }
            Mine.addReplacer("player_prefix") { Mine.toChatMessage(
                VaultAPI.getChat().getPlayerPrefix(it)) }
            Mine.addReplacer("player_suffix") { Mine.toChatMessage(
                Mine.toChatMessage(VaultAPI.getChat().getPlayerPrefix(it))) }
            Mine.addReplacer("group_prefix") {
                Mine.toChatMessage(
                        VaultAPI.getChat().getGroupPrefix("null", VaultAPI.getPermission().getPrimaryGroup(it)))
            }
            Mine.addReplacer("group_suffix") {
                Mine.toChatMessage(
                        VaultAPI.getChat().getGroupSuffix("null", VaultAPI.getPermission().getPrimaryGroup(it)))
            }
            Mine.addReplacer("player_money") {
                if (VaultAPI.hasVault() && VaultAPI.hasEconomy()) {
                    return@addReplacer Extra.MONEY.format(VaultAPI.getEconomy().getBalance(it))
                }
                "0.00"
            }
            Mine.addReplacer("player_balance") {
                if (VaultAPI.hasVault() && VaultAPI.hasEconomy()) {
                    return@addReplacer Extra.formatMoney(VaultAPI.getEconomy().getBalance(it))
                }
                "0.00"
            }
        }
        Mine.addReplacer("players_online") { Mine.getPlayers().size }
        Mine.addReplacer("player_world") { it.world.name }
        Mine.addReplacer("player_display_name") { it.displayName }
        Mine.addReplacer("player_name") { it.name }
        Mine.addReplacer("player_health") { it.health.format() }
        Mine.addReplacer("player_max_health") { it.maxHealth.format() }
        Mine.addReplacer("player_level") { it.level }
        Mine.addReplacer("player_xp") { it.totalExperience.format()}
        Mine.addReplacer("player_kills") { it.getStatistic(Statistic.PLAYER_KILLS) }
        Mine.addReplacer("player_deaths") { it.getStatistic(Statistic.DEATHS) }
        Mine.addReplacer("player_kdr") { p ->
            val kill = p.getStatistic(Statistic.PLAYER_KILLS)
            val death = p.getStatistic(Statistic.DEATHS)
            if (kill == 0||death == 0) return@addReplacer 0
            kill / death
        }
        Mine.addReplacer("player_kill/death") { player ->
            val kill = player.getStatistic(Statistic.PLAYER_KILLS)
            val death = player.getStatistic(Statistic.DEATHS)
            if (kill == 0||death == 0) return@addReplacer 0
            kill / death
        }
        Mine.addReplacer("player_loc_x") { it.location.x }
        Mine.addReplacer("player_loc_y") { it.location.y }
        Mine.addReplacer("player_loc_z") { it.location.z }
    }
}